### PR TITLE
Support --timeout flag to override default 10 minute timeout

### PR DESCRIPTION
**Pull request by mbarrien**
This PR adds a --timeout flag to czecs upgrade, czecs install, and czecs task to allow specifying how long to wait for the service to stabilize/task to complete before considering the operation a failure.

The previous implementation used the default values from the underlying AWS SDK libraries. (see https://github.com/aws/aws-sdk-go/blob/v1.15.47/service/ecs/waiters.go#L82 and https://github.com/aws/aws-sdk-go/blob/v1.15.47/service/ecs/waiters.go#L199). This PR overrides the number of attempts/delay time with our own values/function. We continue to use the original defaults of 10 minutes total (15 seconds between checks for services, 6 seconds between checks for tasks), but we add code to vary the max attempts and last loop delay based on the command line argument.

Internally, we change runTask/deployUpgrade/deployInstall/rollbackInstall from functions to class methods to prevent having to pass around so many arguments. We also move these waiter utilities to util/waiter.go since these functions are shared between the upgrade/install/task commands.

### CODE REVIEW

File: cmd/install.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ type installCmd struct {
 	rollback          bool
 	service           string
 	taskDefinitionArn string
+	timeout           int
```
Comment (edulop91) [inline cmd/install.go]: I would actually make this a time.Duration. 
you can then do things like time.Duration.Parse and it does things like 10m 5s 10h etc

Comment (edulop91) [reply cmd/install.go]: not having a unit associated with a time makes it hard to reason about

File: util/waiter.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	}
+	waiter.SleepWithContext = func(context aws.Context, duration time.Duration) error {
+		log.Debugf("Sleeping, previous response: %+#v", req.Data)
+		return aws.SleepWithContext(context, duration)
+	}
+}
+
+// WaiterDelay returns the WaiterOptions to be able to delay a given amount of seconds,
+// checking if the operation is done every defaultDelay seconds
+func WaiterDelay(timeout int, defaultDelay int) []request.WaiterOption {
```
Comment (edulop91) [inline util/waiter.go]: suggest making these time.Durations

File: cmd/install.go (hunk 1 above)
Comment (mbarrien) [reply cmd/install.go]: I feel hesitant about having to explain to the user the GoLang string duration format strings, and dealing with ms/us/ns which I really don't care about and have no way to enforce. I feel it's simple just to say it's all seconds.

**Pull request merged**